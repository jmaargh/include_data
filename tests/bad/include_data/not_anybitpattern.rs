fn main() {}

static NOT_ABP: char = include_data::include_data!("../../test_data/binary_4");

// This type satisfies the requirements of `AnyBitPattern`, but is not marked as such
#[repr(transparent)]
struct Foo(u16);

static NOT_ABP_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
