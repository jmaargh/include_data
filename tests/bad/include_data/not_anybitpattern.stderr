error[E0277]: the trait bound `char: Pod` is not satisfied
 --> tests/bad/include_data/not_anybitpattern.rs:3:24
  |
3 | static NOT_ABP: char = include_data::include_data!("../../test_data/binary_4");
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Pod` is not implemented for `char`
  |
  = help: the following other types implement trait `Pod`:
            ()
            ManuallyDrop<T>
            Option<T>
            PhantomData<T>
            PhantomPinned
            Wrapping<T>
            [T; N]
            __m128
          and $N others
  = note: required because of the requirements on the impl of `AnyBitPattern` for `char`
note: required by a bound in `NOT_ABP::typecheck`
 --> tests/bad/include_data/not_anybitpattern.rs:3:24
  |
3 | static NOT_ABP: char = include_data::include_data!("../../test_data/binary_4");
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `NOT_ABP::typecheck`
  = note: this error originates in the macro `include_data::include_data` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Foo: Pod` is not satisfied
 --> tests/bad/include_data/not_anybitpattern.rs:9:30
  |
9 | static NOT_ABP_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Pod` is not implemented for `Foo`
  |
  = help: the following other types implement trait `Pod`:
            ()
            ManuallyDrop<T>
            Option<T>
            PhantomData<T>
            PhantomPinned
            Wrapping<T>
            [T; N]
            __m128
          and $N others
  = note: required because of the requirements on the impl of `AnyBitPattern` for `Foo`
note: required by a bound in `NOT_ABP_CUSTOM::typecheck`
 --> tests/bad/include_data/not_anybitpattern.rs:9:30
  |
9 | static NOT_ABP_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `NOT_ABP_CUSTOM::typecheck`
  = note: this error originates in the macro `include_data::include_data` (in Nightly builds, run with -Z macro-backtrace for more info)
